File: dart/space-age/lib/space_age.dart
import 'dart:math';

class SpaceAge {
  var planets = {
		'Mercury': 0.2408467,
		'Venus': 0.61519726,
		'Earth': 1.0,
		'Mars': 1.8808158,
		'Jupiter': 11.862615,
		'Saturn': 29.447498,
		'Uranus': 84.016846,
		'Neptune': 164.79132
	};
  int earthYear = 31557600; // seconds

  double roundDouble(double value, int places) { 
    double mod = pow(10.0, places) as double; 
    return ((value * mod).round().toDouble() / mod); 
  }

  double age({required String planet, required int seconds}) {
    double planetYear = planets[planet] ?? -1;
    double planetAge = seconds / (planetYear * earthYear);
    return roundDouble(planetAge, 2);
  } 
}

File: c/space-age/space_age.c
#include "space_age.h"

float age(planet_t planet, int64_t seconds) 
{
    if (planet < 0 || planet > NEPTUNE) {
        return -1.0f;
    }
    
    float orbital_periods[] = {
        0.2408467f,  // MERCURY
        0.61519726f, // VENUS
        1.0f,        // EARTH
        1.8808158f,  // MARS
        11.862615f,  // JUPITER
        29.447498f,  // SATURN
        84.016846f,  // URANUS
        164.79132f   // NEPTUNE
    };
    
    float earth_years = seconds / 31557600.0f;
    return earth_years / orbital_periods[planet];
}
